django/db/migrations/serializer.py

```diff
@@ -386,6 +386,28 @@ def unregister(cls, type_):
         cls._registry.pop(type_)
 
 
+class GenericAliasSerializer(BaseSerializer):
+    """
+    Generic Serializer class to convert
+    any type into a string
+    """
+
+    def serialize(self):
+        imports = set()
+        value_repr = repr(self.value)
+
+        if hasattr(self.value, "__origin__"):
+            origin = self.value.__origin__
+            if (
+                origin is not None
+                and hasattr(origin, "__module__")
+                and origin.__module__ != "builtins"
+            ):
+                imports.add(f"Import {origin.__module__}")
```

Comment:
This doesn't produce a valid import when serializing this:

`dict[str, models.QuerySet]`

It gives:
```
from django.db import migrations, models
...
                validators=[
                    models.SchemaValidator(
                        dict[str, django.db.models.query.QuerySet]
                    )
                ]
```

Instead, I think we can just delegate the args to the existing serializers:
```py
    def serialize(self):
        imports = set()
        # Avoid iterating self.value, because it returns itself.
        # https://github.com/python/cpython/issues/103450
        for item in self.value.__args__:
            _, item_imports = serializer_factory(item).serialize()
            imports.update(item_imports)
        return repr(self.value), imports
```